compiler-plugin/src/main/kotlin/it/unibo/collektive/frontend/checkers/NoAlignInsideLoop.kt

```diff
@@ -15,38 +20,176 @@ import org.jetbrains.kotlin.fir.expressions.FirWhileLoop
  */
 object NoAlignInsideLoop : FirFunctionCallChecker(MppCheckerKind.Common) {
 
-    private val safeOperators = listOf("alignedOn", "align", "dealign")
+    private val SAFE_OPERATORS = listOf(
+        "it.unibo.collektive.aggregate.api.Aggregate.alignedOn",
+        "it.unibo.collektive.aggregate.api.Aggregate.align",
+        "it.unibo.collektive.aggregate.api.Aggregate.dealign"
+    )
 
-    private fun CheckerContext.isInsideALoopWithoutAlignedOn(): Boolean {
-        val loopElementIndex = containingElements.indexOfLast { it is FirWhileLoop }
-        val functionDeclarationIndex = containingElements.dropLast(1).indexOfLast {
-            (it as? FirSimpleFunction)?.receiverParameter?.isAggregate(session) == true
-        }
-        if (loopElementIndex == -1 || functionDeclarationIndex == -1) {
-            return false
-        }
-        return loopElementIndex > functionDeclarationIndex &&
-            containingElements
-                .drop(loopElementIndex)
-                .dropLast(1)
-                .filterIsInstance<FirFunctionCall>()
-                .map { it.calleeReference.name.asString() }
-                .none { it == "alignedOn" }
-    }
+    /**
+     * Methods used inside collections to iterate their elements.
+     */
+    private val ITERATIVE_METHODS = setOf(
```

Comment:
To prevent an explicit enumeration, what do you think about something like:

```kotlin
    val members = listOf(
        Class.forName("kotlin.collections.CollectionsKt").kotlin,
        Collection::class,
        Iterable::class,
        List::class,
        Map::class,
        Sequence::class,
        Set::class,
    )
        .flatMap { it.java.methods.mapNotNull { it.kotlinFunction } + it.members }
        .filter {
            fun KParameter.isFunctionType(): Boolean = (type.classifier as? KClass<*>)?.qualifiedName
                ?.startsWith("kotlin.Function")
                ?: false
            it.parameters.any { parameter ->
                parameter.isFunctionType()
            }
        }
        .map { it.name }
        .toSet()
```

Should be adjusted, but I think it is a good starting point

Reply:
Thank you! I will try using it inside the checker

Reply:
This method seems to work, but it gave me some issues with the CI.
```
java.lang.UnsupportedOperationException: Packages and file facades are not yet supported in Kotlin reflection. Meanwhile please use Java reflection to inspect this class: class kotlin.collections.CollectionsKt
```
[Here is the failed run](https://github.com/FreshMag/collektive-plugin/actions/runs/11541727746/job/32124073493?pr=12).